pkg/storage/util/util.go

```diff
@@ -104,3 +105,48 @@ func GetValueFromSecret(sec *corev1.Secret, key string) (string, error) {
 	}
 	return "", fmt.Errorf("secret %q does not contain required key %q", fmt.Sprintf("%s/%s", sec.Namespace, sec.Name), key)
 }
+
+// GenerateStorageName generates a unique name for the storage
+// medium that the registry will use
+func GenerateStorageName(listers *regopclient.Listers, additionalInfo ...string) (string, error) {
+
+	// Get the infrastructure name
+	infra, err := GetInfrastructure(listers)
+	if err != nil {
+		return "", err
+	}
+
+	// A slice to store the parts of our name
+	var parts []string
+
+	// Put the infrastructure name first
+	parts = append(parts, infra.Status.InfrastructureName)
+
+	// Image Registry Name second
+	parts = append(parts, imageregistryv1.ImageRegistryName)
+
+	// Additional information provided to the function third
+	parts = append(parts, additionalInfo...)
```

Comment:
For Swift additionalInfo is empty, so the generated name looks like "mfedosin-g7p5v-image-registry--myuprdgckxphuptgcelojvpgaphavyp" with double "-". Not sure if this was done intentionally, but I suggest to append additionalInfo to the parts only if it's not empty.

Reply:
Thanks, I didn't catch that, will fix